### ISSUE TITLE
docker-outside-of-docker feature fails due to signature invalid

### ISSUE THREAD

**Issue by bigtallcampbell** [devcontainers/features#419]
The docker-outside-of-docker feature fails to install when using a dotnet6.0 based container.

Error log:
```bash
[dev_containers_target_stage 3/3] RUN cd /tmp/build-features/docker-outside-of-docker_1 && chmod +x ./devcontainer-features-install.sh && ./devcontainer-features-install.sh:
#0 0.369 ===========================================================================
#0 0.369 Feature       : Docker (docker-outside-of-docker)
#0 0.369 Description   : Re-use the host docker socket, adding the Docker CLI to a container. Feature invokes a script to enable using a forwarded Docker socket within a container to run Docker commands.
#0 0.369 Id            : ghcr.io/devcontainers/features/docker-outside-of-docker
#0 0.369 Version       : 1.0.9
#0 0.369 Documentation : https://github.com/devcontainers/features/tree/main/src/docker-outside-of-docker
#0 0.369 Options       :
#0 0.369     VERSION="latest"
#0 0.369     MOBY="true"
#0 0.369     DOCKERDASHCOMPOSEVERSION="v1"
#0 0.369     ENABLENONROOTDOCKER="true"
#0 0.369 ===========================================================================
#0 0.856 DOCKER_MOBY_ARCHIVE_VERSION_CODENAMES=buster bullseye bionic focal jammy
[2023-01-24T22:58:44.089Z] #0 0.856 Distro codename  'bullseye'  matched filter  'buster bullseye bionic focal jammy'
#0 0.857 MICROSOFT_GPG_KEYS_URI=https://packages.microsoft.com/keys/microsoft.asc
#0 0.974 Get:1 http://deb.debian.org/debian bullseye InRelease [116 kB]
#0 0.979 Get:2 https://packages.microsoft.com/repos/microsoft-debian-bullseye-prod bullseye InRelease [10.5 kB]
#0 0.979 Get:3 https://dl.yarnpkg.com/debian stable InRelease [17.1 kB]
#0 0.992 Get:4 http://deb.debian.org/debian-security bullseye-security InRelease [48.4 kB]
#0 1.027 Get:5 https://packages.microsoft.com/repos/microsoft-debian-bullseye-prod bullseye/main amd64 Packages [110 kB]
#0 1.047 Get:6 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
#0 1.103 Get:7 http://deb.debian.org/debian bullseye/main amd64 Packages [8183 kB]
#0 1.136 Err:3 https://dl.yarnpkg.com/debian stable InRelease
#0 1.136   The following signatures were invalid: EXPKEYSIG 23E7166788B63E1E Yarn Packaging <yarn@dan.cx>
#0 1.214 Get:8 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [216 kB]
#0 1.289 Get:9 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [14.6 kB]
#0 1.992 Reading package lists...
[2023-01-24T22:58:44.089Z] #0 2.331 W: GPG error: https://dl.yarnpkg.com/debian stable InRelease: The following signatures were invalid: EXPKEYSIG 23E7166788B63E1E Yarn Packaging <yarn@dan.cx>
#0 2.331 E: The repository 'https://dl.yarnpkg.com/debian stable InRelease' is not signed.
#0 2.332 ERROR: Feature "Docker (docker-outside-of-docker)" (ghcr.io/devcontainers/features/docker-outside-of-docker) failed to install! Look at the documentation at https://github.com/devcontainers/features/tree/main/src/docker-outside-of-docker for help troubleshooting this error.
------
```

Devcontainer.Json snippet:
```json
"features": {
		"ghcr.io/devcontainers/features/docker-outside-of-docker:1": {
			"version": "latest",
			"enableNonRootDocker": "true",
			"moby": "true"
		}
	}
```

Dockerfile:
```
ARG DOTNET_CONTAINER_VARIANT="6.0-bullseye-slim"
FROM mcr.microsoft.com/vscode/devcontainers/dotnet:0-${DOTNET_CONTAINER_VARIANT}
#FROM mcr.microsoft.com/dotnet/sdk:6.0

```

Changing base container to `mcr.microsoft.com/dotnet/sdk:6.0` does work

**Comment by jkeech** [comment]
Looks like this is related to Yarn's GPG key expiring. Closing as a duplicate of https://github.com/devcontainers/images/issues/370 which is tracking rebuilding and releasing new images with an updated Yarn key.

**Comment by bigtallcampbell** [comment]
Confirmed - the fix in [devcontainers/images/issues/370](https://github.com/devcontainers/images/issues/370) resolved the issue and devcontainers are able to build.  For others, I dumped my images, containers, and cache first, then rebuilt.

```bash
docker system prune --all --force
```

**Comment by rudi1602** [comment]
It is still not working after pruning every cache.
**devcontainer.json**
```
...  
"dockerComposeFile": "../compose.yml",
...
"features": {
    "ghcr.io/devcontainers/features/docker-outside-of-docker:1": {}  
}
...
```
**compose.yml**
```
services:
 app:
  build:
   dockerfile: .devcontainer/Dockerfile
   context: .
  command: sleep infinity
```
**Dockerfile**
```
FROM devcontainers/java:1-21-bullseye
...
```
**Log**
```
...
4.094 Readi
[2026-02-04T14:09:01.594Z] ng package lists...
5.047 W: GPG error: https://dl.yarnpkg.com/debian stable InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 62D54FD4003F6525
5.047 E: The repository 'https://dl.yarnpkg.com/debian stable InRelease' is not signed.
5.057 ERROR: Feature "Docker (docker-outside-of-docker)" (ghcr.io/devcontainers/features/docker-outside-of-docker) failed to install! Look at the documentation at https://github.com/devcontainers/features/tree/main/src/docker-outside-of-docker for help troubleshooting this error.
------
failed to solve: process "/bin/sh -c cp -ar /tmp/build-features-src/docker-outside-of-docker_0 /tmp/dev-container-features  && chmod -R 0755 /tmp/dev-container-features/docker-outside-of-docker_0  && cd /tmp/dev-container-features/docker-outside-of-docker_0  && chmod +x ./devcontainer-features-install.sh  && ./devcontainer-features-install.sh  && rm -rf /tmp/dev-container-features/docker-outside-of-docker_0" did not complete successfully: exit code: 100
[2026-02-04T14:09:01.607Z] Stop (21925 ms): Run: docker compose --project-name myapp -f /workspaces/myapp/compose.yml -f /tmp/devcontainercli-root/docker-compose/docker-compose.devcontainer.build-1770214119680.yml build
[2026-02-04T14:09:01.632Z] Error: Command failed: docker compose --project-name myapp -f /workspaces/myapp/compose.yml -f /tmp/devcontainercli-root/docker-compose/docker-compose.devcontainer.build-1770214119680.yml build
[2026-02-04T14:09:01.632Z]     at lw (/root/.vscode-remote-containers/dist/dev-containers-cli-0.437.0/dist/spec-node/devContainersSpecCLI.js:432:525)
[2026-02-04T14:09:01.633Z]     at async l6 (/root/.vscode-remote-containers/dist/dev-containers-cli-0.437.0/dist/spec-node/devContainersSpecCLI.js:432:2475)
[2026-02-04T14:09:01.633Z]     at async c6 (/root/.vscode-remote-containers/dist/dev-containers-cli-0.437.0/dist/spec-node/devContainersSpecCLI.js:412:3860)
[2026-02-04T14:09:01.633Z]     at async Y6 (/root/.vscode-remote-containers/dist/dev-containers-cli-0.437.0/dist/spec-node/devContainersSpecCLI.js:484:4015)
[2026-02-04T14:09:01.633Z]     at async BC (/root/.vscode-remote-containers/dist/dev-containers-cli-0.437.0/dist/spec-node/devContainersSpecCLI.js:484:4957)
[2026-02-04T14:09:01.633Z]     at async p7 (/root/.vscode-remote-containers/dist/dev-containers-cli-0.437.0/dist/spec-node/devContainersSpecCLI.js:665:202)
[2026-02-04T14:09:01.633Z]     at async d7 (/root/.vscode-remote-containers/dist/dev-containers-cli-0.437.0/dist/spec-node/devContainersSpecCLI.js:664:14804)
[2026-02-04T14:09:01.633Z]     at async /root/.vscode-remote-containers/dist/dev-containers-cli-0.437.0/dist/spec-node/devContainersSpecCLI.js:484:1188
[2026-02-04T14:09:01.805Z] Stop (112669 ms): Run in container: node /root/.vscode-remote-containers/dist/dev-containers-cli-0.437.0/dist/spec-node/devContainersSpecCLI.js up --container-session-data-folder /tmp/devcontainers-12a04f4d-e230-460a-8e21-1861a05bf70d1770213852497 --workspace-folder /workspaces/myapp --workspace-mount-consistency cached --gpu-availability detect --id-label vsch.local.repository=https://mygit/myapp.git --id-label vsch.local.repository.volume=myapp-c00abf90157bb27e0596115094580d9b593636aab650cb485e8ee271bf5f22a2 --id-label vsch.local.repository.folder=myapp --id-label devcontainer.config_file=/workspaces/myapp/.devcontainer/devcontainer.json --log-level debug --log-format json --config /workspaces/myapp/.devcontainer/devcontainer.json --override-config /tmp/devcontainer-f8fb140b-a93f-413d-9dae-643ef214f237.json --default-user-env-probe loginInteractiveShell --mount type=volume,source=myapp-c00abf90157bb27e0596115094580d9b593636aab650cb485e8ee271bf5f22a2,target=/workspaces,external=true --mount type=volume,source=vscode,target=/vscode,external=true --skip-post-create --update-remote-user-uid-default off --mount-workspace-git-root --terminal-columns 157 --terminal-rows 19 --include-configuration --include-merged-configuration
[2026-02-04T14:09:01.805Z] Exit code 1
...
```

Any ideas?

**Workaround**
https://github.com/devcontainers/images/issues/370#issuecomment-1402753345
